cpp/ql/test/library-tests/ir/ir/ir.cpp

```diff
@@ -1219,4 +1219,19 @@ void switch2Case_default(int x) {
     int z = y;
 }
 
+int staticLocalInit(int x) {
+    static int a = 0;  // Constant initialization
+    static int b = sizeof(x);  // Constant initialization
```

Comment:
Looking at my implementation of `runtimeExprInStaticInitializer`, I don't understand why we (correctly) see this as a constant initialization. But somehow we do.

Reply:
`runtimeExprInStaticInitializer()` checks `e.getFullyConverted().isConstant()`, which should certainly hold for a `sizeof(x)`.

Reply:
Okay, looking at it again I understand it better now. It only propagates non-constness up through `getAChild` for `AggregateLiteral`s. An initializer is assumed to be 0 or more levels of aggregate literals at the top and then some other kinds of expression below that, and only the top-level constness of those non-aggregate expressions matter. This means `inStaticInitializer` could be made smaller and faster because it currently contains things like the `x` in `sizeof(x)` for no good reason.

Reply:
See https://github.com/Semmle/ql/pull/3085.